include/cereal/types/memory.hpp

```diff
@@ -325,20 +325,20 @@ namespace cereal
   typename std::enable_if<!traits::has_load_and_construct<T, Archive>::value, void>::type
   CEREAL_LOAD_FUNCTION_NAME( Archive & ar, memory_detail::PtrWrapper<std::shared_ptr<T> &> & wrapper )
   {
-    auto & ptr = wrapper.ptr;
-
     uint32_t id;
 
     ar( CEREAL_NVP_("id", id) );
 
     if( id & detail::msb_32bit )
     {
-      ptr.reset( detail::Construct<T, Archive>::load_andor_construct() );
+      auto  ptr = std::const_pointer_cast<std::decay_t<T>>(wrapper.ptr);
```

Comment:
What I mean is that `ptr` is `reset` in the line right after, so there's no point in assigning it a value, is there?